v2/internal/testcommon/vcr/redact.go

```diff
@@ -109,6 +109,17 @@ func HideRecordingData(azureIDs creds.AzureIDs, s string) string {
 	return s
 }
 
+func HideRecordingDataWithCustomRedaction(azureIDs creds.AzureIDs, s string, redact map[string]string) string {
+	// Replace and hide all the custom data
+	if redact != nil {
+		for k, v := range redact {
+			s = strings.ReplaceAll(s, k, v)
+		}
+	}
+
+	return HideRecordingData(azureIDs, s)
```

Comment:
Do we ever just call `HideRecordingData` now? If not make it unexported possibly? Or just inline it in here and combine these two methods if nobody ever calls it except this?

Reply:
We do call it for v1.vcr recorder. Since we don't use it anymore for recordings so no point supporting redactions there.

Reply:
`v1.vcr` doesn't have a recorder - BUT the v1 player does need access to redactions, because we redact URLs. If we don't have those, we can't match the URLs for playback from the recording.